src/main/java/org/veupathdb/service/eda/ss/model/varcollection/FloatingPointVarCollection.java

```diff
@@ -3,8 +3,9 @@
 import java.util.List;
 import org.veupathdb.service.eda.ss.model.distribution.NumberDistributionConfig;
 import org.veupathdb.service.eda.ss.model.variable.FloatingPointVariable;
+import org.veupathdb.service.eda.ss.model.variable.NumberVariable;
 
-public class FloatingPointVarCollection extends VarCollection<Double, FloatingPointVariable> {
+public class FloatingPointVarCollection<T extends Number & Comparable<T>> extends VarCollection<T, NumberVariable<T>> {
```

Comment:
it seems like a comment explaining that the complexity here is to support integers would be helpful.  otherwise it is kinda confusing, no?